### PR TITLE
Disable parallel join build with disk spilling is triggered

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Parallel join build build has issue when spilling is triggered,
see issue for more details: https://github.com/facebookincubator/velox/issues/3567
Temporarily disable the parallel join when spilling is triggered
and re-enable it after the fix.

### CODE REVIEW

File: velox/common/memory/MemoryAllocator.h [hunk 1]
```diff
@@ -385,24 +385,6 @@ class MemoryAllocator : public std::enable_shared_from_this<MemoryAllocator> {
     uint64_t size_{0};
   };
 
-  /// Stats on memory allocated by allocateBytes().
-  struct AllocateBytesStats {
```
Comment (mbasmanova) [inline velox/common/memory/MemoryAllocator.h:389]: This change seems unrelated. It would be cleaner to submit it as a separate PR.

File: velox/exec/HashBuild.cpp [hunk 2]
```diff
@@ -735,10 +735,14 @@ bool HashBuild::finishHashBuild() {
         }
       }
 
-      const bool hasOthers = !otherTables.empty();
+      // TODO: we need to disable parallel join after
```
Comment (mbasmanova) [inline velox/exec/HashBuild.cpp:738]: Looks like there are some typos. Would you double check?

File: velox/exec/HashBuild.cpp [hunk 3]
```diff
@@ -735,10 +735,14 @@ bool HashBuild::finishHashBuild() {
...
+      // https://github.com/facebookincubator/velox/issues/3567 is fixed.
+      const bool allowParallelJoin =
```
Comment (mbasmanova) [inline velox/exec/HashBuild.cpp:740]: allowParallelJoin - this name is confusing; I believe the parallelism applies only to building hash table, no?

File: velox/exec/tests/JoinFuzzer.cpp [hunk 4]
```diff
@@ -41,7 +41,7 @@ DEFINE_double(
     "Chance of adding a null value in a vector "
     "(expressed as double from 0 to 1).");
 
-DEFINE_bool(enable_spill, false, "Whether to test plans with spilling enabled");
+DEFINE_bool(enable_spill, true, "Whether to test plans with spilling enabled");
```
Comment (mbasmanova) [inline velox/exec/tests/JoinFuzzer.cpp:44]: Should we make this change in a separate PR?

**Review by mbasmanova — COMMENTED** [review]
@xiaoxmeng Some questions and comments.

File: velox/exec/HashBuild.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/exec/HashBuild.cpp:738]: Updated

File: velox/exec/HashBuild.cpp (hunk 3 above)
Comment (xiaoxmeng) [reply velox/exec/HashBuild.cpp]: Yeah, thanks for pointing out. Updated.

File: velox/exec/HashBuild.cpp [hunk 5]
```diff
@@ -735,10 +735,14 @@ bool HashBuild::finishHashBuild() {
         }
       }
 
-      const bool hasOthers = !otherTables.empty();
+      // TODO: we need to disable parallel join build before
```
Comment (mbasmanova) [inline velox/exec/HashBuild.cpp:738]: nit: "TODO: Re-enable after xxx is fixed" since to-do is to re-enable, not to disable :-)

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Thanks.
